### PR TITLE
Document how to check for the existence of current figure/axes.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Summary

This is an alternate approach for #20773, relying only on doc improvements.

## PR Checklist

- [ ] Has pytest style unit tests (and `pytest` passes).
- [ ] Is [Flake 8](https://flake8.pycqa.org/en/latest/) compliant (run `flake8` on changed files to check).
- [ ] New features are documented, with examples if plot related.
- [ ] Documentation is sphinx and numpydoc compliant (the docs should [build](https://matplotlib.org/devel/documenting_mpl.html#building-the-docs) without error).
- [ ] Conforms to Matplotlib style conventions (install `flake8-docstrings` and run `flake8 --docstring-convention=all`).
- [ ] New features have an entry in `doc/users/next_whats_new/` (follow instructions in README.rst there).
- [ ] API changes documented in `doc/api/next_api_changes/` (follow instructions in README.rst there).

### CODE REVIEW

File: lib/matplotlib/figure.py [hunk 1]
```diff
@@ -1484,7 +1484,12 @@ def sca(self, a):
     @docstring.dedent_interpd
     def gca(self, **kwargs):
         """
-        Get the current Axes, creating one if necessary.
+        Get the current Axes.
+
+        If no current Axes exists, a new one is created using `.add_subplot`.
+        (To test whether an Axes exists on a Figure, check whether
+        ``figure.axes`` is empty.  To test whether a current Figure exists,
```
Comment (jklymak) [inline lib/matplotlib/figure.py:1491]:
```suggestion
        ``figure.axes`` is empty.  To test whether any figures exist,
```
I appreciate that you mean a "current figure" as in a figure on the figure stack, but that technical distinction reads strangely, and I don't think needs to be made here.  Ditto below....

Comment (anntzer) [reply lib/matplotlib/figure.py:1491]: I don't have a strong preference, but note that the current wording parallels the first sentence ("if no current Axes exists").  Would you change that too?

Comment (jklymak) [reply lib/matplotlib/figure.py:1491]: No, because `gca` stands for "get current axes"....

Comment (anntzer) [reply lib/matplotlib/figure.py:1491]: But that's the same as gcf() which stands for `get current figure`?

Comment (jklymak) [reply lib/matplotlib/figure.py:1491]: This is not the docs for `gcf`.   You could be more technical and say "To test whether there are any figures in the `gcf()` stack,"  But the phrase "to test whether a current Figure exists" just reads a typo/bad grammar with no context. 

If we want we could change "Get the current Axes from the pyplot Axes stack" to be more explicit as well.  (and maybe replace "stack" by something more correct?)

Comment (anntzer) [reply lib/matplotlib/figure.py]: Ah, I think I see the problem.  I pushed a rewording, does that look better.

**Review by jklymak — APPROVED** [review]

File: lib/matplotlib/figure.py (hunk 1 above)
Comment (jklymak) [reply lib/matplotlib/figure.py]: looks good to me.   This could go in regardless of  #20773...

**Comment by QuLogic** [comment]
> /home/circleci/project/lib/matplotlib/pyplot.py:docstring of matplotlib.pyplot.gca:4: WARNING: more than one target found for cross-reference 'add_subplot': matplotlib.figure.Figure.add_subplot, matplotlib.figure.FigureBase.add_subplot, matplotlib.figure.SubFigure.add_subplot

**Comment by anntzer** [comment]
hopefully fixed now

**Pull request merged**